src/lib/support/DefaultStorageKeyAllocator.h

```diff
@@ -190,6 +190,15 @@ class DefaultStorageKeyAllocator
 
     // Event number counter.
     static StorageKeyName IMEventNumber() { return StorageKeyName::FromConst("g/im/ec"); }
+
+    // Scene Storage
+    static StorageKeyName SceneFabricList() { return StorageKeyName::FromConst("g/sfl"); }
+    static StorageKeyName FabricScenesKey(chip::FabricIndex fabric) { return StorageKeyName::Formatted("f/%x/s", fabric); }
+    static StorageKeyName FabricSceneKey(chip::FabricIndex fabric, chip::EndpointId endpoint, chip::GroupId group,
+                                         chip::SceneId scene)
+    {
+        return StorageKeyName::Formatted("f/%x/s/e%xg%xs%x", fabric, endpoint, group, scene);
```

Comment:
This will require significant work to erase entries per fabric if any issue arises with storage. The key is quite a bit complicated. Could there be a single flat table of scenes per fabric? Basically, a single level with all scenes. You can then walk each entry. If you do it like this, you need very tricky logic to account for errors during multiple-index traversal.